### PR TITLE
Complete redesign and refactor

### PR DESCRIPTION
**Pull request by r1chm8**

### CODE REVIEW

File: src/components/FolderManager.vue [hunk 1]
```diff
@@ -1,74 +1,61 @@
...
+                </button>
+            </div>
+        </form>
     </modal>
 </template>
 
 <script>
-import { mapGetters, mapMutations } from 'vuex';
+import { mapActions, mapGetters } from 'vuex';
+import { actions as apiActions } from '../index';
```
Comment (Jack97) [inline src/components/FolderManager.vue:40]: The apiActions alias could probably have a better name

File: src/components/MediaManager.vue [hunk 2]
```diff
@@ -1,280 +1,197 @@
...
 export default {
     components: {
         Breadcrumb,
         Confirmation,
-        Dropdown,
         Modal,
 
-        Folders,
         Media,
+        Folders,
```
Comment (Jack97) [inline src/components/MediaManager.vue:102]: Media and Folders component names are not too descriptive

File: src/components/MediaUploader.vue [hunk 3]
```diff
@@ -175,33 +187,19 @@ export default {
         },
 
         generateUuid() {
```
Comment (Jack97) [inline src/components/MediaUploader.vue:189]: Quite an expensive method to generate a unique id, could potentially be simplified - i.e by using an incrementing integer.

File: src/components/partials/Media.vue [hunk 4]
```diff
@@ -1,75 +1,169 @@
...
+                </span>
+            </a>
+        </div>
+
+        <transition name="mm-fade">
+            <div v-if="! isLoading && ! currentMedia.length" class="mm-notification">
+                This folder is empty.
+            </div>
+
+            <div v-else-if="! isLoading && currentMedia.length" class="mm-media-holder">
```
Comment (Jack97) [inline src/components/partials/Media.vue:25]: Could move this logic to a method / computed property

File: src/components/ui/Confirmation.vue [hunk 5]
```diff
@@ -40,31 +44,87 @@ export default {
...
 
     methods: {
         ...mapActions({
-            deleteFocusedItems: 'mediaManager/deleteFocusedItems',
-        }),
-
-        ...mapMutations({
             close: 'mediaManager/closeConfirmation',
+            removeFolders: 'mediaManagerFolders/removeFolders',
+            removeMediaItems: 'mediaManagerMedia/remove',
```
Comment (Jack97) [inline src/components/ui/Confirmation.vue:99]: Inconsistent naming

File: src/util/api-client.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+const client = axios.create({
```
Comment (Jack97) [inline src/util/api-client.js:1]: Not sure about the api-client name.

**Review by Jack97 — COMMENTED** [review]
Generally all looks really good, I'd recommend the following changes along with the comments I made:

* The vuex actions, getters, mutations should be more verbose / consistent. Don't rely on the namespace to describe what the method does.

* Try and move logic out of the template to make the components easier to understand at a glance. Will also help with consistency and dry the components up a bit.

File: src/components/FolderManager.vue (hunk 1 above)
Comment (r1chm8) [reply src/components/FolderManager.vue:40]: Leaving this for the time being

File: src/util/api-client.js (hunk 6 above)
Comment (r1chm8) [reply src/util/api-client.js:1]: Leaving this for the time being

**Pull request merged**